packages/react-priority-overflow/src/useOverflowItem.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import { useOverflowContext } from './overflowContext';
+
+/**
+ * Registers an overflow item
+ * @param id - unique identifier for the item used by the overflow manager
+ * @param priority - higher priority means the item overflows later
+ * @param groupId - assigns the item to a group, group visibility can be watched
+ * @returns ref to assign to an intrinsic HTML element
+ */
+export function useOverflowItem<TElement extends HTMLElement>(id: string, priority?: number, groupId?: string) {
+  const ref = React.useRef<TElement>(null);
+  const registerItem = useOverflowContext(v => v.registerItem);
+
+  useIsomorphicLayoutEffect(() => {
+    if (ref.current) {
+      return registerItem({
+        element: ref.current,
+        id: id + '',
```

Comment:
```suggestion
        id,
```

Reply:
ah yeah, I didn't remove the actual casts 😅😅